include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -99,4 +99,15 @@ def MemoryOp : CalyxPrimitive<"memory", [
       "ArrayRef<int64_t>":$addrSizes
     )>
   ];
+
+  let extraClassDeclaration = [{
+    size_t getNumAddresses() { return addrSizes().size(); }
+
+    ValueRange addrPorts() { return getResults().take_front(getNumAddresses()); }
+    Value addr(size_t i) { assert(i<getNumAddresses()); return getResults()[i]; }
```

Comment:
Add a message to assert `... && "Out of bounds of address ports for this memory."`